Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM node:10
+# docker build -t sourcecred .
+# available at vanessa/sourcecred if you don't want to build
+RUN apt-get update && \
+    apt-get install -y apt-transport-https python
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+    echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
+    apt-get update && \
+    apt-get install -y yarn build-essential && \
+    mkdir -p /code
+WORKDIR /code
+ADD . /code
+RUN curl --compressed -o- -L https://yarnpkg.com/install.sh | bash && \
+    yarn && \
+    yarn backend
+ENTRYPOINT ["/bin/bash", "/code/scripts/docker-entrypoint.sh"]
```

Comment:
And if unset, where does it go (where is it going now?). It would be good to have a consistent export location, and then we can give instruction to the user to bind as a volume. If it doesn't matter, I'll set it as SOURCECRED_DIRECTORY=/data.

Reply:
And I'll do a build arg so some future user can change it.

Reply:
> And if unset, where does it go (where is it going now?). It would be good to have a consistent export location, and then we can give instruction to the user to bind as a volume. If it doesn't matter, I'll set it as SOURCECRED_DIRECTORY=/data.

The default I believe is `${PWD}/sourcecred_data` meaning it would start writing it to a new layer of the container.

What's the advantage of a build arg over env + volume here?

Reply:
The build arg just means that if someone wants to re-build the container, they can provide the `--build-arg`  flag for some custom value. It will default to what we specified.

Reply:
Docker always has one writable overlay layer (hence why you can write) and it would use it for the default or /data, regardless of the path I think. Singularity would be another story - we wouldn't have writable at all and would need to specify an overlay or bind to the host.

Reply:
Oh oups, I need to set the build arg to be the environment variable, totally goofed that. :)

Reply:
okay I fixed my goof! Just pushed a change so now the variable is export:
```bash
$ docker run -it --entrypoint bash sourcecred
root@a28e47d07cad:/code# echo $SOURCECRED_DIRECTORY 
/data
```